client/services/FeedbackGeneratorService.js

```diff
@@ -415,6 +415,20 @@ tie.factory('FeedbackGeneratorService', [
             'Please keep your code within the existing predefined functions',
             '-- we cannot process code in the global scope.'
           ].join(' '));
+        } else if (prereqCheckFailure.hasWrongLanguage()) {
+          WRONG_LANGUAGE_ERRORS.python.forEach(function(error) {
+            if (error.errorName === prereqCheckFailure.getWrongLangKey()) {
+              error.feedbackParagraphs.forEach(function(json) {
+                if (json._type === 'text') {
```

Comment:
Shouldn't be referencing private stuff here. Declare methods on the object if you need to.

Also, "json" is not very descriptive as a variable name. Maybe `paragraph` (if that's what this is)?